qa/tasks/cephfs/fuse_mount.py

```diff
@@ -424,20 +425,24 @@ def find_socket(client_name):
         for f in files:
                 pid = re.match(".*\.(\d+)\.asok$", f).group(1)
                 if os.path.exists("/proc/{{0}}".format(pid)):
-                        return f
+                    with open("/proc/{{0}}/cmdline".format(pid), 'r') as proc_f:
+                        contents = proc_f.read()
+                        if mountpoint in contents:
```

Comment:
Wh6y is this change necessary?

Reply:
This might be redundant. I'll look into this and get back to you.

Reply:
The machine where vstart_runner.py is run might have asok files from previous run and this part of code might pick one of those files instead. When this happens, we would have incorrect PID. So, I added code to look for mountpoint in cmdline file to confirm that PID is from current run and not a previous one.

I was wondering why this never happened before. Turns out it was simply luck. I printed few message from within `find_socket()` to be able to see that. PID of new processes are higher than the previous ones and therefore they turn out to be the very first element in the sequence `files` (which the code loops over here). Following is the output  from a run on my machine that confirms this -
```
files = ['/tmp/ceph-asok.qfxklY/client.0.79206.asok', '/tmp/ceph-asok.qfxklY/client.0.76357.asok', '/tmp/ceph-asok.qfxklY/client.0.74666.asok', '/tmp/ceph-asok.qfxklY/client.0.73630.asok', '/tmp/ceph-asok.qfxklY/client.0.72759.asok', '/tmp/ceph-asok.qfxklY/client.0.68885.asok', '/tmp/ceph-asok.qfxklY/client.0.68065.asok', '/tmp/ceph-asok.qfxklY/client.0.26756.asok']
mountpoint = /tmp/tmpj8w3adt3/mnt.0
file = /tmp/ceph-asok.qfxklY/client.0.79206.asok
/tmp/ceph-asok.qfxklY/client.0.79206.asok
2020-05-26 15:24:02,158.158 INFO:tasks.cephfs.fuse_mount:Found client admin socket at /tmp/ceph-asok.qfxklY/client.0.79206.asok
> /home/rishabh/repos/ceph/master/qa/tasks/vstart_runner.py(751)mount()
-> log.info("Mounting client.{0} with pid {1}".format(self.client_id, self.fuse_daemon.subproc.pid))
```

So I think we should keep this check to avoid being right just due to this coincidence.